employees/tests.py

```diff
@@ -288,142 +290,302 @@ def test_should_month_update_view_work(self):
+                6: [True, date(2018, 3, 19)],
+                12: [True, date(2017, 5, 2)],
+                11: [False, None],
+                29: [False, None],
+                }
+
+        for key in dict.keys():
+            from employees.views_business_logic import ContractExtension
+            contractExtension = ContractExtension()
+
+            extend_status = dict.get(key)[0]
+
+            self.assertEqual(contractExtension.add_three_months(key), extend_status)
+            exp_date = dict.get(key)[1]
+
+            self.assertEqual(contractExtension.exp_date, exp_date)
+
+
+# class ContractExtensionViewStatusTests(TestCase):
+#     fixtures = ['users_myuser.json', 'employees_employee.json', 'month.json']
+#
+#     # fixtures = ['fikstura_cities','users_myuser.json', 'employees_employee.json']
+#     # cities deleted manual from fixtury. To test with cities activate upper line and use employees_employee_with_cities.json
+#     # reason- loading fikstura_cities takes very long time
+#
+#     def setUp(self):
+#         self.client.login(username='pawel.kisielewicz@codepoets.it', password='codepoets')
+#         zbigniew_adamski = Employee.objects.get(id=3)
+#         assert ('Zbigniew Adamski' == str(
+#             zbigniew_adamski)), 'Zbigniew Adamski should be in fixtures, check if file contains Zbigniew Adamski or problem with loading fixtures'
+#
+#     def test_should_business_logic_extension_work_properly(self):
+#         dict = {3: True,
+#                 2: True,
+#                 4: True,
+#                 6: True,
+#                 12: True,
+#                 11: False,
+#                 29: False
+#                 }
+#
+#         for key in dict.keys():
+#             self.__should_contract_extension_work_properly(key, dict.get(key), 'add_1_month')
+#             self.__should_contract_extension_work_properly(key, dict.get(key), 'add_3_month')
+#
+#     def __should_contract_extension_work_properly(self, id, answer_status, extension):
+#
+#         kwargs = {'HTTP_X_REQUESTED_WITH': 'XMLHttpRequest'}
+#         url = reverse('employee_action')
+#
+#         get_data = {'extensionLength': extension,
+#                     'employeeId': id,
+#                     }
+#
+#         response = self.client.post(url, get_data, **kwargs)
+#         self.assertEqual(response.status_code, 200)
+#
+#         import json
+#         dict = json.loads(response.content)
+#
+#         status = dict[0]
+#         name = dict[1]
+#         date = dict[2]
+#
+#         self.assertEqual(status, answer_status)
+
+
+class ContractExtensionViewCorrectDateTests(TestCase):
+    fixtures = ['users_myuser.json', 'employees_employee.json', 'month.json']
+
+    # fixtures = ['fikstura_cities','users_myuser.json', 'employees_employee.json']
+    # cities deleted manual from fixtury. To test with cities activate upper line and use employees_employee_with_cities.json
+    # reason- loading fikstura_cities takes very long time
```

Comment:
same here

Reply:
look again, its not dead code. I start tests without cities with edited file employees (deleted cities). It works good and i left message why i did it. Fixed to "nice" comment. Is it OK?